### PR TITLE
[charts] Fix text position in Safari

### LINKED ISSUE #10808: [charts] tick position displayed incorrectly in Safari
### Steps to reproduce

Link to live example: [mui.com](https://mui.com/x/react-charts/lines/#data-format)

Steps:
1. Use Safari

### Current behavior

In Safari
Bildschirmfoto 2023-10-26 um 09 55 13

### Expected behavior

Like in Chrome
Bildschirmfoto 2023-10-26 um 09 52 29

### Context

_No response_

### Your environment

_No response_

**Search keywords**: tick position, safari, charts

### PR DESCRIPTION
**Pull request by lhilgert9**
Fixes #10808.

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Thanks to @alexfauquette.

This fix can be reverted when Apple release a new Safari version with the fix mentioned at
[https://webkit.org/blog/14390/release-notes-for-safari-technology-preview-174/](https://webkit.org/blog/14390/release-notes-for-safari-technology-preview-174/).

> Fixed dominant-baseline CSS property to be inherited ([265525@main](https://commits.webkit.org/265525@main))

### CODE REVIEW

File: packages/x-charts/src/internals/components/ChartsText.tsx [hunk 1]
```diff
@@ -93,7 +93,12 @@ export function ChartsText(props: ChartsTextProps) {
       style={style}
     >
       {wordsByLines.map((line, index) => (
-        <tspan x={x} dy={`${index === 0 ? startDy : wordsByLines[0].height}px`} key={index}>
+        <tspan
+          x={x}
+          dy={`${index === 0 ? startDy : wordsByLines[0].height}px`}
+          dominantBaseline={dominantBaseline}
```
Comment (alexfauquette) [inline packages/x-charts/src/internals/components/ChartsText.tsx:99]:
```suggestion
          dominantBaseline={dominantBaseline} // Propagated to fix Safari issue: https://github.com/mui/mui-x/issues/10808
```

I thought about adding it in the code such that in a year can still see why it's here

**Review by alexfauquette — APPROVED** [review]
Thanks :)

**Comment by lhilgert9** [comment]
@alexfauquette I hit a wrong button and try to fix it and now I think I made it even worse. Can you help me please. I have accidentally merged the main branch into my branch and then try to undo it but it don't work. Hope you can help me.

**Comment by lhilgert9** [comment]
Okay, I think I've got it. Not the prettiest way, but it seems to work.

**Pull request merged**